lorawan-encoding/src/creator.rs

```diff
@@ -294,10 +287,10 @@ impl<D: AsMut<[u8]>, F: CryptoFactory> JoinRequestCreator<D, F> {
     /// # Argument
     ///
     /// * key - the key to be used for setting the MIC.
-    pub fn build(&mut self, key: &AppKey) -> &[u8] {
+    pub fn build<F: CryptoFactory>(&mut self, key: &AppKey, factory: &F) -> &[u8] {
         let d = self.data.as_mut();
-        set_mic(d, &key.0, &self.factory);
-        d
+        set_mic(&mut d[..JOIN_REQUEST_LEN], &key.0, factory);
+        &d[..JOIN_REQUEST_LEN]
```

Comment:
This fixes a breaking bug with my last PR :/ Without this the whole buffer is returned, not just what belongs to this creator. There is one more like this